SU2_CFD/src/fluid/CDataDrivenFluid.cpp

```diff
@@ -268,11 +334,14 @@ void CDataDrivenFluid::Evaluate_Dataset(su2double rho, su2double e) {
   }
 }
 
-void CDataDrivenFluid::Run_Newton_Solver(su2double Y1_target, su2double Y2_target, su2double* Y1, su2double* Y2,
-                                         su2double* dY1drho, su2double* dY1de, su2double* dY2drho, su2double* dY2de) {
+void CDataDrivenFluid::Run_Newton_Solver(su2double Y1_target, su2double Y2_target, su2double& Y1, su2double& Y2,
+                                         su2double& dY1drho, su2double& dY1de, su2double& dY2drho, su2double& dY2de) {
```

Comment:
Why are some of these passed by value and others by non-const reference? Are they outputs?
Why aren't they preaccumulated either as input or output?

Reply:
They could indeed be passed by const value and const reference, since none of them are outputs. I still need to verify whether I did the preaccumulation correctly.